hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java

```diff
@@ -842,6 +853,20 @@ private Set<String> getDeltaFiles(OmSnapshot fromSnapshot,
     return deltaFiles;
   }
 
+  private void validateEstimatedKeyChangesAreInLimits(
+      ManagedSstFileReader sstFileReader
+  ) throws RocksDBException, IOException {
+    if (sstFileReader.getEstimatedTotalKeys() >
+        maxAllowedKeyChangesForASnapDiff) {
+      throw new IOException(
```

Comment:
Nit: We can do this as part of a different PR, should we have a separate exception for Snapshots?

Reply:
Yes, I'll add a TODO.

May be we can do it as part of [HDDS-8202(https://issues.apache.org/jira/browse/HDDS-8202).

Reply:
Added a TODO: https://github.com/apache/ozone/pull/4617/files#diff-46f82d5fbac5dc7aae1fc363811fb88e6bc6c89a2aefafddad90de2f1ff3cfbdR944